src/geo/leaflet/vector.js

```diff
@@ -0,0 +1,95 @@
+var cdb = window.cdb;
+var L = cdb.L;
+var _ = cdb._;
+var util = cdb.core.util;
+var LeafletLayerView = cdb.geo.LeafletLayerView;
+var CartoDBLogo = cdb.geo.common.CartoDBLogo;
+
+var LeafletVectorLayer = L.CartoDBd3Layer.extend({
+
+  initialize: function(layerModel, leafletMap) {
+    var extra = layerModel.get('extra_params');
+
+    var query = this._getQuery(layerModel);
+
+    // initialize the base layers
+    L.CartoDBd3Layer.prototype.initialize.call(this, {
+      table: layerModel.get('table_name'),
+      user: layerModel.get('user_name'),
+      blendmode: layerModel.get('torque-blend-mode'),
+      sql_api_domain: layerModel.get('sql_api_domain'),
+      sql_api_protocol: layerModel.get('sql_api_protocol'),
+      sql_api_port: layerModel.get('sql_api_port'),
+      tiler_protocol: layerModel.get('tiler_protocol'),
+      tiler_domain: layerModel.get('tiler_domain'),
+      tiler_port: layerModel.get('tiler_port'),
+      maps_api_template: layerModel.get('maps_api_template'),
+      stat_tag: layerModel.get('stat_tag'),
+      animationDuration: layerModel.get('torque-duration'),
+      steps: layerModel.get('torque-steps'),
+      sql: query,
+      visible: layerModel.get('visible'),
+      extra_params: {
+        api_key: extra ? extra.map_key: ''
+      },
+      cartodb_logo: layerModel.get('cartodb_logo'),
+      attribution: layerModel.get('attribution'),
+      cartocss: layerModel.get('cartocss') || layerModel.get('tile_style'),
+      named_map: layerModel.get('named_map'),
+      auth_token: layerModel.get('auth_token'),
+      no_cdn: layerModel.get('no_cdn'),
+      dynamic_cdn: layerModel.get('dynamic_cdn'),
+      loop: layerModel.get('loop') === false? false: true,
+      instanciateCallback: function() {
+        var cartocss = layerModel.get('cartocss') || layerModel.get('tile_style');
+
+        return '_cdbct_' + cdb.core.util.uniqueCallbackName(cartocss + query);
+      }
+    });
+
+    cdb.geo.LeafletLayerView.call(this, layerModel, this, leafletMap);
+
+    // match leaflet events with backbone events
+    this.fire = this.trigger;
+
+    this.bind('tilesLoaded', function() {
+      this.trigger('load');
+    }, this);
+
+    this.bind('tilesLoading', function() {
+      this.trigger('loading');
+    }, this);
+
+  },
+
+  onAdd: function(map) {
+    L.CartoDBd3Layer.prototype.onAdd.apply(this, [map]);
+    // Add CartoDB logo
+    if (this.options.cartodb_logo != false)
+      cdb.geo.common.CartoDBLogo.addWadus({ left:8, bottom:8 }, 0, map._container)
+  },
+
+  _getQuery: function(layerModel) {
+    var query = layerModel.get('query');
+    var qw = layerModel.get('query_wrapper');
+    if(qw) {
+      query = _.template(qw)({ sql: query || ('select * from ' + layerModel.get('table_name')) });
```

Comment:
Do we want to keep support for `table_name`? Or should we make mandatory to have a SQL?

cc @javisantana